CMakeLists.txt

```diff
@@ -334,30 +354,24 @@ if(CMAKE_USE_OPENSSL)
 endif()
 
 if(NOT CURL_DISABLE_LDAP)
-  if(WIN32)
-    option(USE_WIN32_LDAP "Use Windows LDAP implementation" ON)
-    if(USE_WIN32_LDAP)
-      check_library_exists_concat("wldap32" cldap_open HAVE_WLDAP32)
-      if(NOT HAVE_WLDAP32)
-        set(USE_WIN32_LDAP OFF)
-      endif()
-    endif()
+  CMAKE_DEPENDENT_OPTION(USE_WIN32_LDAP "Use Windows LDAP implementation" ON
+    "WIN32;USE_WINDOWS_SSPI" OFF)
+  if(USE_WIN32_LDAP)
+    check_library_exists_concat("wldap32" cldap_open HAVE_WLDAP32)
+    check_include_file_concat("winldap.h" HAVE_WINLDAP_H)
+    check_include_file_concat("winber.h"  HAVE_WINBER_H)
+    set(USE_WIN32_LDAP HAVE_WLDAP32 AND HAVE_WINLDAP_H AND HAVE_WINBER_H)
```

Comment:
The `set` command will not evaluate the conditions.  Use an explicit `if` to set `USE_WIN32_LDAP`.